kingpin/actors/aws/cloudformation.py

```diff
@@ -0,0 +1,434 @@
+
+        Raises:
+            StackNotFound: If the stack doesn't exist.
+        """
+        while True:
+            stack = yield self._get_stack(self.option('name'))
+
+            if not stack:
+                msg = 'Stack %s not found?' % self.option('name')
+                raise StackNotFound(msg)
+
+            self.log.debug('Got stack %s status: %s' %
+                           (stack.stack_name, stack.stack_status))
+
+            # First, lets see if the stack is still in progress (either
+            # creation, deletion, or rollback .. doesn't really matter)
+            if stack.stack_status in IN_PROGRESS:
+                self.log.info('Stack is in %s, waiting %s(s)...' %
+                              (stack.stack_status, sleep))
+                yield utils.tornado_sleep(sleep)
+                continue
+
+            # If the stack is in the desired state, then return
+            if stack.stack_status in desired_states:
+                self.log.info('Stack execution completed, final state: %s' %
+                              stack.stack_status)
+                raise gen.Return()
+
+            # Lastly, if we get here, then something is very wrong and we got
+            # some funky status back. Throw an exception.
+            msg = 'Unxpected stack state received (%s)' % stack.stack_status
+            raise CloudFormationError(msg)
+
+
+class Create(CloudFormationBaseActor):
+
+    """Creates an Amazon CF Stack.
+
+    http://boto.readthedocs.org/en/latest/ref/cloudformation.html
+    #boto.cloudformation.connection.CloudFormationConnection.create_stack
+
+    """
+
+    all_options = {
+        'capabilities': (list, [],
+                         'The list of capabilities that you want to allow '
+                         'in the stack'),
+        'disable_rollback': (bool, False,
+                             'Set to `True` to disable rollback of the stack '
+                             'if stack creation failed.'),
+        'name': (str, REQUIRED, 'Name of the stack'),
+        'parameters': (dict, {}, 'Parameters passed into the CF '
+                                 'template execution'),
+        'region': (str, REQUIRED, 'AWS region name, like us-west-2'),
+        'template': (str, REQUIRED,
+                     'Path to the AWS CloudFormation File. http(s)://, '
+                     'file:///, absolute or relative file paths.'),
+        'timeout_in_minutes': (int, 60,
+                               'The amount of time that can pass before the '
+                               'stack status becomes CREATE_FAILED'),
+    }
+
+    def __init__(self, *args, **kwargs):
+        """Initialize our object variables."""
+        super(Create, self).__init__(*args, **kwargs)
+
+        # Check if the supplied CF template is a local file. If it is, read it
+        # into memory.
+        (self._template_body, self._template_url) = self._get_template_body(
```

Comment:
Not sure why this needs to be in the `__init__` and not just in `_execute()`.

Also add a comment here that this is for boto's `validate_template` otherwise it's confusing why you're preparing these.